sktime/pipeline/_graph_pipe.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements forecaster for selecting among different model classes."""
+
+import pandas as pd
+from sklearn.base import clone
+
+from sktime.base import BaseEstimator
+from sktime.base._simple_base import SimpleBaseEstimator, SimpleBaseObject
+from sktime.base._tests_and_checks import LengthChecker
+from sktime.forecasting.base import BaseForecaster
+from sktime.transformations.base import BaseTransformer
+
+__author__ = ["miraep8"]
+__all__ = ["GraphOutput", "GraphNode", "GraphPipeline"]
+
+
+class GraphOutput:
```

Comment:
Is the idea of this class to manage and collect the results of all graphnodes and steps? 

I like this idea, this probably makes the pipeline class easier to understand.